tests/integration/site-details/same-activity-dates/same-activity-dates.test.js

```diff
@@ -0,0 +1,220 @@
+import { JSDOM } from 'jsdom'
+import { getByRole, getByText } from '@testing-library/dom'
+import { createServer } from '~/src/server/index.js'
+import { statusCodes } from '~/src/server/common/constants/status-codes.js'
+import {
+  getExemptionCache,
+  updateExemptionMultipleSiteDetails
+} from '~/src/server/common/helpers/session-cache/utils.js'
+
+jest.mock('~/src/server/common/helpers/session-cache/utils.js')
+
+describe('Same activity dates page', () => {
+  let server
+
+  const mockExemption = {
+    id: 'test-exemption-123',
+    projectName: 'Test Project',
+    multipleSiteDetails: {
+      multipleSitesEnabled: true
+    }
+  }
+
+  jest.mocked(getExemptionCache).mockReturnValue(mockExemption)
+  jest.mocked(updateExemptionMultipleSiteDetails).mockReturnValue({})
+
+  beforeAll(async () => {
+    server = await createServer()
+    await server.initialize()
+  })
+
+  afterAll(async () => {
+    await server.stop()
+  })
```

Comment:
Maybe we could extract these to a `serverTestSetup()` function that returns the server instance, will save a lot of duplicated code in tests?